src/modules/Game.class.js

```diff
@@ -1,68 +1,263 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
-  }
-
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  static gameStatuses = {
+    idle: 'idle',
+    playing: 'playing',
+    win: 'win',
+    lose: 'lose',
+  };
+
+  constructor(
+    initialState = Array.from({ length: 4 }, () => Array(4).fill(0)),
+  ) {
+    this.initialState = initialState;
+    this.state = initialState.map((row) => [...row]);
+    this.status = Game.gameStatuses.idle;
+    this.score = 0;
+  }
+
+  moveLeft() {
+    if (!this.isStateValid(this.state)) {
+      return;
+    }
+
+    const updatedState = this.state.map((row) => this.applyMove(row));
+
+    this.updateGameState(updatedState);
+    this.completeMoveTasks();
+  }
```

Comment:
The `moveLeft` method does not check if the move actually changes the state before completing the move tasks. This could result in generating new tiles even when the board hasn't changed, which is not typical for a 2048 game. Consider adding a check to see if the state has changed after applying the move.